exonum/src/node/mod.rs

```diff
@@ -837,3 +814,48 @@ impl Node {
         ApiSender::new(self.channel.api_requests.0.clone())
     }
 }
+
+/// Public for testing
+#[doc(hidden)]
+pub fn create_public_api_handler(
+    blockchain: Blockchain,
+    pool: TxPool,
+    shared_api_state: SharedNodeState,
+    config: &NodeApiConfig,
+) -> Chain {
+    let mut mount = Mount::new();
+    mount.mount("api/services", blockchain.mount_public_api());
+
+    if config.enable_blockchain_explorer {
+        let mut router = Router::new();
+        let explorer_api = public::ExplorerApi::new(blockchain.clone());
+        explorer_api.wire(&mut router);
+        mount.mount("api/explorer", router);
+    }
+
+    let mut router = Router::new();
+    let system_api = public::SystemApi::new(pool, blockchain, shared_api_state);
+    system_api.wire(&mut router);
+    mount.mount("api/system", router);
+
+    Chain::new(mount)
```

Comment:
Maybe, a stupid question, but: what purpose does wrapping a `Mount` into a `Chain` serve? It seems that the code would work without it (including a potential change of the return type into `impl Handler` in the future).

Reply:
The real reason is that I've just copied what was originally written. However, I do think that returning a `Chain` here is the right thing to do: this allows us to add middlewares without changing the signature. This, for example, will happen with CORS: https://github.com/exonum/exonum/pull/406.

Reply:
Makes sense, thanks! As far as I get from the Iron docs, `Chain` is kind of like `Box<Handler>` in that it could be wrapped around any handler hiding its implementation details (and allowing to add middleware for the handler).